### ISSUE TITLE
[Bug]: failed jobs due to random "Error UNKNOWN while writing to socket. Connection lost"

### ISSUE THREAD

**Issue by klaemo** [taskforcesh/bullmq#2542]
### Version

vpy2.7.2

### Platform

Python

### What happened?

We have a python worker that consumes jobs from a nodejs queue. It mostly works really well, but we do get random `Error UNKNOWN while writing to socket. Connection lost` errors on the python side. On the nodejs side we do not observer these connection losses.

Connection losses are fine and expected in a distributed system, but I'm surprised they bubble up to the job and make it fail 🤔 As far as I can tell there is some kind of retry logic in place also in the python client. But, does it work? Shouldn't the gracefully handle temporary connectivity issues?

- python `3.11.5`
- redis `7.0.15`, hosted on DigitalOcean
- the worker started with all the default settings, we only specify the redis connection details

### How to reproduce.

I am not sure. We're unable to reliably reproduce this.

### Relevant log output

```shell
ERROR: Error UNKNOWN while writing to socket. Connection lost.
Traceback (most recent call last):
  File "/usr/src/app/.venv/lib/python3.11/site-packages/redis/asyncio/connection.py", line 473, in send_packed_command
    await self._writer.drain()
  File "/usr/local/lib/python3.11/asyncio/streams.py", line 378, in drain
    await self._protocol._drain_helper()
  File "/usr/local/lib/python3.11/asyncio/streams.py", line 167, in _drain_helper
    raise ConnectionResetError('Connection lost')
ConnectionResetError: Connection lost

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/app/.venv/lib/python3.11/site-packages/bullmq/utils.py", line 9, in extract_result
    return job_task.result()
           ^^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/bullmq/worker.py", line 116, in getNextJob
    job_instance = await self.moveToActive(token)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/bullmq/worker.py", line 125, in moveToActive
    result = await self.scripts.moveToActive(token, self.opts)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/bullmq/scripts.py", line 464, in moveToActive
    result = await self.commands["moveToActive"](keys=keys, args=args)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/redis/commands/core.py", line 5172, in __call__
    return await client.evalsha(self.sha, len(keys), *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/redis/asyncio/client.py", line 612, in execute_command
    return await conn.retry.call_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/.venv/lib/python3.11/site-packages/redis/asyncio/client.py", line 585, in _send_command_parse_response
    await conn.send_command(*args)
  File "/usr/src/app/.venv/lib/python3.11/site-packages/redis/asyncio/connection.py", line 497, in send_command
    await self.send_packed_command(
  File "/usr/src/app/.venv/lib/python3.11/site-packages/redis/asyncio/connection.py", line 484, in send_packed_command
    raise ConnectionError(
redis.exceptions.ConnectionError: Error UNKNOWN while writing to socket. Connection lost.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by klaemo** [comment]
It might be that the [retry strategy](https://github.com/taskforcesh/bullmq/blob/master/python/bullmq/redis_connection.py#L27) is way too short: 

```python
retry = Retry(ExponentialBackoff(), 3)
```

The default `ExponentialBackoff` ramps up from 8ms to max 512ms intervals. That seems way to short, especially compared to the JS version which starts at 1s and ramps up to 20s. Also, bullmq limits itself to 3 retries in the python version.

I'm currently experimenting with something closer to the JS version:
```python
retry = Retry(ExponentialBackoff(cap=20, base=1), 20)
```

I'll report back how it goes!

**Comment by klaemo** [comment]
Fixed by #2546
